modxcms/revolution
# Restore download listeners

### What does it do?
Restore the two step download handling removed in #15385. Add a failure listener.

### Why is it needed?
Allow success and failure handling during downloading the file similar to #16152

### How to test
Create a file in a media source, delete that file elsewhere without reloading the browser window. You get a blank page with the error message "Cannot open file: /path/to/the/file". After the patch, you see an error message window.

### Related issue(s)/PR(s)
#16152, #15385
